### ISSUE TITLE
`usage: sphinx-build...` error in the middle of running spelling (similar to #142)

### ISSUE THREAD

**Issue by tgross35** [sphinx-contrib/spelling#180]
## Error

The issue from #142 seems to be present again, I'm coming across it when trying to spellcheck docs for sqlalchemy. It fails somewhere in changelog_07.rst, here's the traceback:

```
find_spec('mssql.TIME') failed, invalid module name: No module named 'mssql'
find_spec('util.py3k') failed, invalid module name: No module named 'util'
find_spec('util.py3k_warning') failed, invalid module name: No module named 'util'
find_spec('Session.dirty') failed, invalid module name: No module named 'Session'
removing .py extension from 'history_meta.py' before searching for module
find_spec('select.bind') failed, invalid module name: __path__ attribute not found on 'select' while trying to find 'select.bind'
find_spec('Inspector.get_table_names') failed, invalid module name: No module named 'Inspector'
find_spec('pg.ENUM') failed, invalid module name: No module named 'pg'
find_spec('Session.is_active') failed, invalid module name: No module named 'Session'
removing .py extension from 'dictlike-polymorphic.py' before searching for module
usage: sphinx-build [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: sphinx-build --help [cmd1 cmd2 ...]
   or: sphinx-build --help-commands
   or: sphinx-build cmd --help

error: option -b not recognized
```

## Environment
```
Sphinx==5.0.1
sphinx-copybutton==0.5.0
sphinx-paramlinks @ git+https://github.com/sqlalchemyorg/sphinx-paramlinks.git@2b488538968c38d4f11b761afa707c2319cd9f2c
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==2.0.0
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.5
sphinxcontrib-spelling==7.5.0
```

Validated within a docker container

## Steps to replicate

Clone my fork that has spelling support added. Script for the whole thing:

```
git clone https://github.com/tgross35/sqlalchemy.git
cd sqlalchemy
git checkout b71dbd # known error at this point
cd doc/build
pip install -r requirements.txt
sphinx-build -b spelling -d output/doctrees -T -j auto . output/spelling -vvv
```

**Comment by dhellmann** [comment]
I've seen this sort of thing before when autodoc is used and a module with code that runs on import tries to parse arguments. I think I've also seen it happen when the config is processed twice, but that may have been a different issue.

**Comment by dhellmann** [comment]
The usage text looked suspicious. I found that text in the copy of `distutils` that `setuptools` vendors [here](https://github.com/pypa/setuptools/blob/main/setuptools/_distutils/core.py#L26). It is used as a `SystemExit` exception [here](https://github.com/pypa/setuptools/blob/main/setuptools/_distutils/core.py#L166) when the `setup()` function is run and has trouble parsing arguments.

Changing the exception thrown at line 166 to a `BaseException` gives this output:

```
find_spec('relationship.order_by') failed, invalid module name: No module named 'relationship'
Traceback (most recent call last):
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 423, in next
    (word, pos) = next(self._curtok)
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 141, in __next__
    return self.next()
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/en.py", line 185, in next
    raise StopIteration()
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 423, in next
    (word, pos) = next(self._curtok)
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 141, in __next__
    return self.next()
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 261, in next
    raise StopIteration()
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 423, in next
    (word, pos) = next(self._curtok)
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 141, in __next__
    return self.next()
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 261, in next
    raise StopIteration()
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 423, in next
    (word, pos) = next(self._curtok)
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 141, in __next__
    return self.next()
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 261, in next
    raise StopIteration()
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/setuptools/_distutils/fancy_getopt.py", line 233, in getopt
    opts, args = getopt.getopt(args, short_opts, self.long_opts)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/getopt.py", line 95, in getopt
    opts, args = do_shorts(opts, args[0][1:], shortopts, args[1:])
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/getopt.py", line 195, in do_shorts
    if short_has_arg(opt, shortopts):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/getopt.py", line 211, in short_has_arg
    raise GetoptError(_('option -%s not recognized') % opt, opt)
getopt.GetoptError: option -b not recognized

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 133, in setup
    ok = dist.parse_command_line()
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 476, in parse_command_line
    args = parser.getopt(args=self.script_args, object=self)
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/setuptools/_distutils/fancy_getopt.py", line 235, in getopt
    raise DistutilsArgError(msg)
distutils.errors.DistutilsArgError: option -b not recognized

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/bin/sphinx-build", line 8, in <module>
    sys.exit(main())
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/sphinx/cmd/build.py", line 290, in main
    return build_main(argv)
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/sphinx/cmd/build.py", line 276, in build_main
    app.build(args.force_all, filenames)
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/sphinx/application.py", line 329, in build
    self.builder.build_update()
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 285, in build_update
    self.build(['__all__'], to_build)
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 352, in build
    self.write(docnames, list(updated_docnames), method)
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 544, in write
    self._write_serial(sorted(docnames))
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 554, in _write_serial
    self.write_doc(docname, doctree)
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/sphinxcontrib/spelling/builder.py", line 191, in write_doc
    lines = list(self._find_misspellings(docname, doctree))
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/sphinxcontrib/spelling/builder.py", line 232, in _find_misspellings
    for (
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/sphinxcontrib/spelling/checker.py", line 54, in check
    for word, pos in self.tokenizer(text):
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 416, in __next__
    return self.next()
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 426, in next
    (word, pos) = next(self._tokenizer)
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 416, in __next__
    return self.next()
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 426, in next
    (word, pos) = next(self._tokenizer)
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 416, in __next__
    return self.next()
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 426, in next
    (word, pos) = next(self._tokenizer)
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 416, in __next__
    return self.next()
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/enchant/tokenize/__init__.py", line 427, in next
    while self._skip(self._to_string(word)):
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/sphinxcontrib/spelling/filters.py", line 211, in _skip
    mod = importlib.util.find_spec(word)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/importlib/util.py", line 94, in find_spec
    parent = __import__(parent_name, fromlist=['__path__'])
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/sqlalchemy/setup.py", line 141, in <module>
    run_setup(False)
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/sqlalchemy/setup.py", line 130, in run_setup
    setup(cmdclass=cmdclass, distclass=Distribution, **kwargs)
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/setuptools/__init__.py", line 87, in setup
    return distutils.core.setup(**attrs)
  File "/Users/dhellmann/Envs/tmp-fecf2f55d59474b/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 135, in setup
    raise BaseException(gen_usage(dist.script_name) + "\nerror: %s" % msg)
BaseException: usage: sphinx-build [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: sphinx-build --help [cmd1 cmd2 ...]
   or: sphinx-build --help-commands
   or: sphinx-build cmd --help

error: option -b not recognized
```

Before the traceback, I also see:

```
writing output... [  4%] changelog/changelog_07
```

The spelling builder code invoked in that stack is the `ImportableModuleFilter` where it tries to import a module [here](https://github.com/sphinx-contrib/spelling/blob/master/sphinxcontrib/spelling/filters.py#L211). Changing the error handling there to look for a `BaseException` suppresses the exit and allows the build to continue past that point.

See #181 for a fix

**Comment by dhellmann** [comment]
The fix for this is in 7.5.1

**Comment by tgross35** [comment]
Excellent, thank you for fixing quick!
